src/NzbDrone.Core/Datastore/Migration/204_add_release_hash.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using Dapper;
+using FluentMigrator;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Datastore.Migration.Framework;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(204)]
+    public class add_add_release_hash : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            Alter.Table("EpisodeFiles").AddColumn("ReleaseHash").AsString().Nullable();
+
+            Execute.WithConnection(UpdateEpisodeFiles);
+        }
+
+        private void UpdateEpisodeFiles(IDbConnection conn, IDbTransaction tran)
+        {
+            var updates = new List<object>();
+
+            using (var cmd = conn.CreateCommand())
+            {
+                cmd.Transaction = tran;
+                cmd.CommandText = "SELECT \"Id\", \"SceneName\", \"RelativePath\", \"OriginalFilePath\" FROM \"EpisodeFiles\" WHERE \"ReleaseHash\" IS NULL";
+
+                using var reader = cmd.ExecuteReader();
+                while (reader.Read())
+                {
+                    var id = reader.GetInt32(0);
+                    var sceneName = reader.IsDBNull(1) ? null : reader.GetString(1);
+                    var relativePath = reader.IsDBNull(2) ? null : reader.GetString(2);
+                    var originalFilePath = reader.IsDBNull(3) ? null : reader.GetString(3);
```

Comment:
These can be simplified to `var x= reader[n] as string;` which correctly handles null values.

Reply:
Good tip. I'm so ingrained in the older pattern, never thought of using that.